src/bootman/bootman.c

```diff
@@ -735,31 +758,62 @@ bool boot_manager_enumerate_initrds_freestanding(BootManager *self)
                 char *initrd_name_key = NULL;
                 char *initrd_name_val = NULL;
                 autofree(char) *path = NULL;
+                struct InitrdEntry *entry = NULL;
 
-                path = string_printf("%s/%s", self->initrd_freestanding_dir, ent->d_name);
+                path = string_printf("%s/%s", dir, ent->d_name);
 
                 /* Some kind of broken link */
-                if (lstat(path, &st) != 0) {
-                        continue;
-                }
+                CHECK_DBG_CONTINUE(lstat(path, &st) != 0, "Broken link: %s, skipping.", path);
 
                 /* Regular only */
-                if (!S_ISREG(st.st_mode)) {
-                        continue;
-                }
+                CHECK_DBG_CONTINUE(!S_ISREG(st.st_mode) && !S_ISLNK(st.st_mode),
+                                   "Non regular file: %s, skipping.", path);
 
                 /* empty files are skipped too */
-                if (st.st_size == 0) {
-                        continue;
-                }
+                CHECK_DBG_CONTINUE(st.st_size == 0, "Empty file: %s, skipping.", path);
 
-                initrd_name_val = strdup(ent->d_name);
-                OOM_CHECK(initrd_name_val);
+                if (S_ISLNK(st.st_mode)) {
+                        char *buf;
+                        ssize_t nbytes, bufsiz;
+
+                        if (st.st_size == 0) {
+                                bufsiz = PATH_MAX;
+                        }
+
+                        buf = alloca((size_t)bufsiz);
+                        OOM_CHECK(buf);
+
+                        nbytes = readlink(path, buf, (size_t)bufsiz);
```

Comment:
Since `bufsiz` is cast to `size_t` twice, can it just be declared with that type instead of `ssize_t`?

Reply:
sure